### ISSUE TITLE
cargo audit --color=never still prints colors

### ISSUE THREAD

**Issue by basvandijk** [RustSec/cargo-audit#177]
`cargo audit --color=never` doesn't respect the `color` setting since colors are still being printed.

This might be because the [`warning_color`](https://github.com/RustSec/cargo-audit/blob/c255b3481f97929037fc11a992df0defa6da07ac/src/presenter.rs#L224:L231) function doesn't take that config flag into account.

**Comment by tarcieri** [comment]
Can you try the latest mater?

The `warning_color` thing is a red herring: the colors should be disabled in the underlying terminal subsystem.

There may have been a bug in the previous release, though.

**Comment by nmattia** [comment]
I've just tried with the latest master (https://github.com/RustSec/cargo-audit/commit/866f46be0f4302dc0073f13c8610ea69577999ea) and the issue is still there. It's not only the warnings, but also things like `Loading`, `Scanning`, `Success` (green) and information about crates (yellow). I'm not sure which `rustsec-crate` I'm depending on though, the `patch` doesn't have a revision set.

@tarcieri what about replacing the `patch` section with a git dependency like this?

``` toml
[dependencies]
...
rustsec = { git = "https://github.com/RustSec/rustsec-crate", rev = "8b2fb1c4cd71e94c8d7d6cdfedc233e418f373a0" }
```

**Comment by tarcieri** [comment]
> I'm not sure which rustsec-crate I'm depending on though, the patch doesn't have a revision set.

Check Cargo.lock. However, it doesn't matter: the color-related functionality is all in `cargo-audit` itself.

I will investigate. It sounds like there's a bug and the proper color config isn't getting passed through to `termcolor`.

> what about replacing the patch section with a git dependency like this?

Those syntaxes are six of one, half dozen of another, however I think `[patch]` better semantically conveys the temporary nature of the patch (i.e. until I can cut another `rustsec` crate release)

**Comment by nmattia** [comment]
> the temporary nature of the patch

Gotcha, didn't get that part! Thanks for looking into this, really appreciate it. FYI I reproduced the issue on Darwin.

**Comment by tjodden** [comment]
Also, `--color=always` does not enable colors when `cargo audit` has it's output redirected to a pipe or file.

**Comment by tarcieri** [comment]
This should be fixed in #199
